src/applications/vaos/tests/appointment-list/components/CommunityCareAppointmentDetailsPage.unit.spec.jsx

```diff
@@ -354,4 +354,102 @@ describe('VAOS <CommunityCareAppointmentDetailsPage>', () => {
 
     expect(screen.getByText(/Community care/)).to.be.ok;
   });
+
+  it('should verify community care calendar ics file format', async () => {
+    const url = '/cc/8a4885896a22f88f016a2cb7f5de0062';
+
+    const appointment = getCCAppointmentMock();
+    appointment.id = '8a4885896a22f88f016a2cb7f5de0062';
+    appointment.attributes = {
+      ...appointment.attributes,
+      appointmentRequestId: '8a4885896a22f88f016a2cb7f5de0062',
+      distanceEligibleConfirmed: true,
+      name: { firstName: 'Rick', lastName: 'Katz' },
+      providerPractice: 'My Eye Dr',
+      providerPhone: '(703) 555-1264',
+      address: {
+        street: '123',
+        city: 'Burke',
+        state: 'VA',
+        zipCode: '20151',
+      },
+      appointmentTime: '05/20/2021 14:15:00',
+      timeZone: 'UTC',
+    };
+
+    mockSingleCommunityCareAppointmentFetch({
+      appointment,
+    });
+    const startDateTime = moment(appointment.attributes.appointmentTime);
+
+    const screen = renderWithStoreAndRouter(
+      <AppointmentList featureHomepageRefresh />,
+      {
+        initialState,
+        path: url,
+      },
+    );
+
+    // Verify page content...
+    expect(
+      await screen.findByRole('heading', {
+        level: 1,
+        name: /^Thursday, May 20, 2021/,
+      }),
+    ).to.be.ok;
+
+    const ics = decodeURIComponent(
+      screen
+        .getByRole('link', {
+          name: 'Add May 20, 2021 appointment to your calendar',
+        })
+        .getAttribute('href')
+        .replace('data:text/calendar;charset=utf-8,', ''),
+    );
+    const tokens = ics.split('\r\n');
```

Comment:
I wish we had a better way of parsing and validating these ics strings, but the only JS [validator library](https://mozilla-comm.github.io/ical.js/) I can find looks like a dead project. Maybe something we can explore in a future ticket.

Reply:
I was thinking the same thing and came up with the code below. I decided not to use it because I prefer to have something more explicit about what it's doing. I really don't like the array indexes. Changing them is a pain. So, I wanted to access the data by key. The code below needs a minor tweak as well to deal with duplicate keys (BEGIN and END) but doable using a hash table. 

```
'BEGIN:VCALENDAR\nVERSION:2.0 PRODID:VA\nBEGIN:VEVENT\nUID:2A6B9C0F-80B9-4A0A-863E-4582BB77C8A6\nSUMMARY:Appointment at CHY PC CASSIDY\nDESCRIPTION:You have a health care appointment at CHY PC CASSIDY\n\t\\n\\n\t2360 East Pershing Boulevard \t\\n\tCheyenne\\, WY 82001-5356 \t\\n\t307-778-7550 \t\\n\\n\tSign in to VA.gov to get details about this appointment\nLOCATION:2360 East Pershing Boulevard\\, Cheyenne\\, WY 82001-5356\nDTSTAMP:20210510T011729Z\nDTSTART:20210510T011729Z\nDTEND:20210510T014729Z\nEND:VEVENT\nEND:VCALENDAR'
    tokens = new Map(
      tokens.map(t => {
        if (t.startsWith('\t')) return [t, t];
        return t.split(':');
      }),
    );
    console.log(tokens);
    console.log(tokens.get('BEGIN'));
```

Reply:
Cool, I think something like that could work and be an improvement. I'll make a ticket to look into both the validator library and approaches like what you started there.